src/llama_index_alloydb_pg/async_chat_store.py

```diff
@@ -0,0 +1,234 @@
+
+        Args:
+            key (object): Key to prevent direct constructor usage.
+            engine (AlloyDBEngine): Database connection pool.
+            table_name (str): Table name that stores the chat store.
+            schema_name (str): The schema name where the table is located.
+                Defaults to "public"
+
+        Raises:
+            Exception: If constructor is directly called by the user.
+        """
+        if key != AsyncAlloyDBChatStore.__create_key:
+            raise Exception("Only create class through 'create' method!")
+
+        # Delegate to Pydantic's __init__
+        super().__init__()
+        self._engine = engine
+        self._table_name = table_name
+        self._schema_name = schema_name
+
+    @classmethod
+    async def create(
+        cls,
+        engine: AlloyDBEngine,
+        table_name: str,
+        schema_name: str = "public",
+    ) -> AsyncAlloyDBChatStore:
+        """Create a new AsyncAlloyDBChatStore instance.
+
+        Args:
+            engine (AlloyDBEngine): AlloyDB engine to use.
+            table_name (str): Table name that stores the chat store.
+            schema_name (str): The schema name where the table is located.
+                Defaults to "public"
+
+        Raises:
+            ValueError: If the table provided does not contain required schema.
+
+        Returns:
+            AsyncAlloyDBChatStore: A newly created instance of AsyncAlloyDBChatStore.
+        """
+        table_schema = await engine._aload_table_schema(table_name, schema_name)
+        column_names = table_schema.columns.keys()
+
+        required_columns = ["id", "key", "message"]
+
+        if not (all(x in column_names for x in required_columns)):
+            raise ValueError(
+                f"Table '{schema_name}'.'{table_name}' has an incorrect schema.\n"
+                f"Expected column names: {required_columns}\n"
+                f"Provided column names: {column_names}\n"
+                "Please create the table with the following schema:\n"
+                f"CREATE TABLE {schema_name}.{table_name} (\n"
+                "    id SERIAL PRIMARY KEY,\n"
+                "    key VARCHAR NOT NULL,\n"
+                "    message JSON NOT NULL\n"
+                ");"
+            )
+
+        return cls(cls.__create_key, engine._pool, table_name, schema_name)
+
+    async def __aexecute_query(self, query, params=None):
+        async with self._engine.connect() as conn:
+            await conn.execute(text(query), params)
+            await conn.commit()
+
+    async def __afetch_query(self, query):
+        async with self._engine.connect() as conn:
+            result = await conn.execute(text(query))
+            result_map = result.mappings()
+            results = result_map.fetchall()
+            await conn.commit()
+        return results
+
+    @classmethod
+    def class_name(cls) -> str:
+        """Get class name."""
+        return "AsyncAlloyDBChatStore"
```

Comment:
nit: (more just an observation)

I am aware that `class_name()` is part of the LlamaIndex `BaseChatStore` interface, however this feels like a silly thing to make every child class to have to do, you would expect the default to just be the class name.

I wonder why LlamaIndex does not just have the `BaseChatStore` do the following:

```python
class BaseChatStore:
# ...
@classmethod
def class_name(cls):
    return cls.__name__
```

Reply:
TIL: Apparently it makes serialization more robust.

[BaseComponent](https://github.com/run-llama/llama_index/blob/dbd89ab7aac3e5de2dcf111454dc037936e59e6b/llama-index-core/llama_index/core/schema.py#L89)
> Get the class name, used as a unique ID in serialization.
>
> This provides a key that makes serialization robust against actual class
> name changes.